app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,49 @@
+class MerchantsController < ApplicationController
+  before_action :set_merchant, only: [:show, :edit, :update, :destroy]
+
+  def index
+    if params[:sort] == 'a-z'
+      @merchants = Merchant.order(:name)
+    elsif params[:sort] == 'z-a'
+      @merchants = Merchant.order(name: :desc)
+    else
+      @merchants = Merchant.all
+    end
+  end
```

Comment:
Nice job using activerecord here. Again, I'd take this into the model, and then organize your merchants there.